diff --git a/node_modules/react-native-ble-peripheral/android/build.gradle b/node_modules/react-native-ble-peripheral/android/build.gradle
index 9b5638c..de0889d 100644
--- a/node_modules/react-native-ble-peripheral/android/build.gradle
+++ b/node_modules/react-native-ble-peripheral/android/build.gradle
@@ -1,12 +1,11 @@
 apply plugin: "com.android.library"
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.3"
+    compileSdkVersion 34
 
     defaultConfig {
         minSdkVersion 23
-        targetSdkVersion 23
+        targetSdkVersion 34
         versionCode 1
         versionName "1.0"
         ndk {
@@ -15,8 +14,8 @@ android {
     }
 }
 dependencies {
-    compile fileTree(dir: "libs", include: ["*.jar"])
-    compile "com.android.support:appcompat-v7:23.0.1"
-    compile "com.facebook.react:react-native:+"  // From node_modules
+    implementation fileTree(dir: "libs", include: ["*.jar"])
+    implementation "com.android.support:appcompat-v7:23.0.1"
+    implementation "com.facebook.react:react-native:+"  // From node_modules
 }
 
diff --git a/node_modules/react-native-ble-peripheral/android/src/main/java/com/himelbrand/ble/peripheral/RNBLEModule.java b/node_modules/react-native-ble-peripheral/android/src/main/java/com/himelbrand/ble/peripheral/RNBLEModule.java
index 71a5f60..7012c45 100644
--- a/node_modules/react-native-ble-peripheral/android/src/main/java/com/himelbrand/ble/peripheral/RNBLEModule.java
+++ b/node_modules/react-native-ble-peripheral/android/src/main/java/com/himelbrand/ble/peripheral/RNBLEModule.java
@@ -22,7 +22,7 @@ import android.content.Context;
 import android.os.Bundle;
 import android.os.Handler;
 import android.os.ParcelUuid;
-import android.support.v7.app.AppCompatActivity;
+import androidx.appcompat.app.AppCompatActivity;
 import android.text.TextUtils;
 import android.util.Log;
 
@@ -52,7 +52,7 @@ import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.bridge.Arguments;
 import com.facebook.react.bridge.WritableArray;
-
+import com.facebook.react.modules.core.DeviceEventManagerModule;
 
 
 /**
@@ -156,6 +156,12 @@ public class RNBLEModule extends ReactContextBaseJavaModule{
             }
             map.putArray("data", data);
             map.putString("device", device.toString());
+            // Emit event to JS
+            if (reactContext != null) {
+                reactContext
+                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                    .emit("onCentralWrite", map);
+            }
             if (responseNeeded) {
                 mGattServer.sendResponse(device, requestId, BluetoothGatt.GATT_SUCCESS, offset, value);
             }
@@ -244,4 +250,13 @@ public class RNBLEModule extends ReactContextBaseJavaModule{
         promise.resolve(this.advertising);
     }
 
+    @ReactMethod
+    public void addListener(String eventName) {
+        // Required for RN built-in Event Emitter
+    }
+
+    @ReactMethod
+    public void removeListeners(Integer count) {
+        // Required for RN built-in Event Emitter
+    }
 }
